### PR TITLE
feat(issue-stream): Update issue stream header design and column breakpoints

### LINKED ISSUE #77824: Issue Stream Table: Decrease horizontal padding and add column separators
Estimate: 0.5 days

Image

### LINKED ISSUE #77825: Issue Stream Table: Support small viewport widths
Progressively hide columns in this order: last + age -> priority -> events + users -> trend -> assignee

Estimate: 0.5 days

### PR DESCRIPTION
**Pull request by MichaelSun48**
closes #77824
closes #77825

This PR makes two major changes to the issue stream (both under the `issue-stream-table-layout` flag):
1. Updates the design of the header to be narrower, to use "Trend" instead of "Graph:", and to have dividers
2. Updates the order at which the columns disappear to reflect the spec in #77825. it is now: last & age -> priority -> events 
 & users -> trend -> assignee

Header design before:

image

Header design after:

image

Column breakpoint behavior: 

https://github.com/user-attachments/assets/9492cfbc-8e64-4c64-b6eb-0ab02d229879

### CODE REVIEW

File: static/app/views/issueList/actions/utils.tsx [hunk 1]
```diff
@@ -161,3 +162,16 @@ export function performanceIssuesSupportsIgnoreAction(
...
+export const COLUMN_BREAKPOINTS = {
+  ISSUE: undefined, // Issue column is always visible
+  TREND: commonTheme.breakpoints.small,
+  AGE: commonTheme.breakpoints.xlarge,
+  SEEN: commonTheme.breakpoints.xlarge,
+  EVENTS: commonTheme.breakpoints.medium,
+  USERS: commonTheme.breakpoints.medium,
+  PRIORITY: commonTheme.breakpoints.large,
+  ASSIGNEE: commonTheme.breakpoints.xsmall,
+};
```
Comment (MichaelSun48) [inline static/app/views/issueList/actions/utils.tsx:177]: I know this adds a bunch of dynamic emotion styles, but it is already a massive pain in the ass to synchronize the margins/spacing between both the header and the rows, we don't need to add column breakpoints to that list, so having this single source of truth is much cleaner.

File: static/app/components/IssueStreamHeaderLabel.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+const IssueStreamHeaderLabel = styled('div')<{breakpoint?: string}>`
+  position: relative;
+  display: inline-block;
+  margin-right: ${space(2)};
+  justify-content: space-between;
+  font-size: ${p => p.theme.fontSizeMedium};
+  font-weight: ${p => p.theme.fontWeightBold};
+  color: ${p => p.theme.subText};
+
+  ${p => p.breakpoint && `@media (max-width: ${p.breakpoint}) { display: none; }`}
```
Comment (scttcper) [inline static/app/components/IssueStreamHeaderLabel.tsx:14]:
```suggestion
  ${p => p.breakpoint && css`@media (max-width: ${p.breakpoint}) { display: none; }`}
```

File: static/app/views/issueList/actions/headers.tsx [hunk 3]
```diff
@@ -142,6 +217,17 @@ const PriorityLabel = styled(ToolbarHeader)<{isSavedSearchesOpen?: boolean}>`
   }
 `;
 
+const NarrowPriorityLabel = styled(IssueStreamHeaderLabel)`
+  display: flex;
+  justify-content: space-between;
+  width: 70px;
+
+  /* prettier-ignore */
```
Comment (scttcper) [inline static/app/views/issueList/actions/headers.tsx:225]: i'm not sure these are ignoring anything and the silly thing we use to process these does not remove comments at build time and they make it into production. might be worth not leaving these comments

File: static/app/views/issueList/actions/index.tsx [hunk 4]
```diff
@@ -135,14 +139,15 @@ function ActionsBarPriority({
                 onUpdate={handleUpdate}
               />
             </HeaderButtonsWrapper>
-          )}
-          {!anySelected && (
+          ) : organization.features.includes('issue-stream-table-layout') ? (
+            <NarrowHeaderButtonsWrapper>
+              <IssueStreamHeaderLabel>{t('Issue')}</IssueStreamHeaderLabel>
```
Comment (malwilley) [inline static/app/views/issueList/actions/index.tsx:144]: These headers look a little big - the designs have them at font-size 13px, but fontSizeSmall is probably also fine

File: static/app/views/issueList/actions/utils.tsx (hunk 1 above)
Comment (malwilley) [reply static/app/views/issueList/actions/utils.tsx:177]: I like this a lot better!

**Review by malwilley — COMMENTED** [review]
Looks great! And looks the same as before when the feature flag is off too

**Review by malwilley — APPROVED** [review]

**Pull request merged**